### PR TITLE
Modcat 221

### PR DESCRIPTION
**Pull request by eturienzo**
## Purpose

<a href="https://issues.folio.org/browse/MODCAT-221">MODCAT-221</a> Get authority record 

## Approach

<ul>
<li>Create an API to get an authority
<ul>
<li>If the authority number exits return the authority record
</ul>
<li>Unit tests updates:
<ul>
<li>Units tests have been added
</ul>
</ul>

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this introduce breaking changes?
  - [x] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [x] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -270,6 +270,18 @@
             "marccat.authority.insert.post"
           ]
         },
+        {
+          "methods": [
+            "GET"
+          ],
+          "pathPattern": "/marccat/authority-record/{id}",
+          "modulePermissions": [
```
Comment (carmentrazza) [inline descriptors/ModuleDescriptor-template.json:278]: Reformat like this:
 "permissionsRequired": ["marccat.authority-record.item.get" ]

File: src/main/java/org/folio/marccat/resources/AuthorityRecordAPI.java [hunk 2]
```diff
@@ -35,6 +37,28 @@
...
+      } else {
+        return new ResponseEntity<>(container, HttpStatus.NOT_FOUND);
+      }
+      return new ResponseEntity<>(container, HttpStatus.OK);
+    }, tenant, okapiUrl, configurator);
+  }
+
   @PostMapping("/authority-record")
   public ResponseEntity<Object> save(@RequestBody final AuthorityRecord record,
     // ContainerRecordTemplate container,
```
Comment (carmentrazza) [inline src/main/java/org/folio/marccat/resources/AuthorityRecordAPI.java:64]: Remove comment

File: src/test/java/org/folio/marccat/resources/AuthorityRecordTest.java [hunk 3]
```diff
@@ -10,46 +10,53 @@
...
       .headers(headers)
       .queryParam("view", "-1")
       .queryParam("lang", "eng")
       .body(templateJson)
       .when()
-      .post(url)
-      .then()
-      .statusCode(201);
+        .post(url);
+    authorityId = myResponse.jsonPath().get("body").toString();
```
Comment (carmentrazza) [inline src/test/java/org/folio/marccat/resources/AuthorityRecordTest.java:52]: Some tests should be done on the authorityId ?

Comment (LeireDiez) [reply src/test/java/org/folio/marccat/resources/AuthorityRecordTest.java:52]: authorityId is used to test the api GET

File: src/main/java/org/folio/marccat/resources/AuthorityRecordAPI.java (hunk 2 above)
Comment (LeireDiez) [reply src/main/java/org/folio/marccat/resources/AuthorityRecordAPI.java]: Removed

File: descriptors/ModuleDescriptor-template.json (hunk 1 above)
Comment (LeireDiez) [reply descriptors/ModuleDescriptor-template.json:278]: Changed

**Review by carmentrazza — APPROVED** [review]

**Review by LeireDiez — APPROVED** [review]

**Pull request merged**